File: app/src/main/java/com/glucocontrol/domain/usecase/reading/GetReadingsByDateRangeUseCase.kt
package com.glucocontrol.domain.usecase.reading

import com.glucocontrol.domain.model.GlucoseReading
import com.glucocontrol.domain.repository.GlucoseRepository
import kotlinx.coroutines.flow.Flow
import java.time.LocalDate
import javax.inject.Inject

/**
 * Devuelve en tiempo real todas las lecturas comprendidas en un rango de fechas.
 *
 * Se usa como base para las consultas de exportación y para filtros personalizados
 * en la pantalla de historial.
 *
 * Devuelve [Flow] para que la UI reaccione automáticamente si los datos cambian
 * mientras la pantalla está activa.
 */
class GetReadingsByDateRangeUseCase
@Inject
constructor(private val repository: GlucoseRepository) {
    operator fun invoke(from: LocalDate, to: LocalDate): Flow<List<GlucoseReading>> {
        require(!from.isAfter(to)) {
            "La fecha de inicio ($from) no puede ser posterior a la fecha de fin ($to)"
        }
        return repository.getReadingsByDateRange(from, to)
    }
}

File: lib/domain/usecase/query/get_readings_by_date_range_use_case.dart
import 'package:glucocontrol/domain/model/glucose_reading.dart';
import 'package:glucocontrol/domain/repository/glucose_repository.dart';

class GetReadingsByDateRangeUseCase {
  final GlucoseRepository _repo;
  GetReadingsByDateRangeUseCase(this._repo);

  Stream<List<GlucoseReading>> call(DateTime from, DateTime to) {
    if (from.isAfter(to)) {
      throw ArgumentError('La fecha de inicio no puede ser posterior a la de fin');
    }
    return _repo.getReadingsByDateRange(from, to);
  }
}
